### PR TITLE
Fix test for core versions above or equal to 2.266

### PR DESCRIPTION
**Pull request by MRamonLeon**
A security test fails for Jenkins versions above 2.266 because on those versions, Jenkins triggers an AccessDeniedException3, instead of the AccessDeniedException2.

This PR fixes this test.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by MRamonLeon** [comment]
@tykeal we need this in to the release.

**Comment by tykeal** [comment]
@MRamonLeon that's fine by me. I'm just happy my change finally got merged ;)

I did a quick look at the checks failing and they're all failing on setting up the JDK. I dunno what's up with that, but it seems to only be the GitHub Actions checks. Jenkins checks themselves all passed fine.

File: src/test/java/org/jenkinsci/plugins/configfiles/Security2203Test.java [hunk 1]
```diff
@@ -142,8 +143,8 @@ private void assertWhoCanExecute(Runnable run, Permission permission, String che
         try (ACLContext ctx = ACL.as(User.getOrCreateByIdOrFullName("reader"))) {
             run.run(); // The method should fail
             fail(String.format("%s should be only accessible by people with the permission %s, but it's accessible by a person with %s", checkedMethod, permission, Item.READ));
-        } catch (AccessDeniedException2 e) {
-            assertThat(e.permission, equalTo(permission));
+        } catch (AccessDeniedException  e) {
```
Comment (alecharp) [inline src/test/java/org/jenkinsci/plugins/configfiles/Security2203Test.java:146]:
```suggestion
        } catch (AccessDeniedException e) {
```

**Comment by MRamonLeon** [comment]
> @MRamonLeon that's fine by me. I'm just happy my change finally got merged ;)
> 
> I did a quick look at the checks failing and they're all failing on setting up the JDK. I dunno what's up with that, but it seems to only be the GitHub Actions checks. Jenkins checks themselves all passed fine.

Yes, it's because GH changes, I think I fixed that. In addition there is a bug with the 3.7.1 that prevents you to change the plugin configuration from the global tool configuration. Wait for next release. See: https://github.com/jenkinsci/config-file-provider-plugin/releases/tag/config-file-provider-3.7.1

**Review by rsandell — APPROVED** [review]

**Review by alecharp — APPROVED** [review]
Unimportant suggestion. It's all good!

**Pull request merged**